src/ResponseEngine/tests/ResponseEngineTest.php

```diff
@@ -360,6 +360,59 @@ public function testWebChatButtonMessageWithTabSwitchButton()
         $this->assertInstanceOf('OpenDialogAi\ResponseEngine\Message\Webchat\WebchatButtonMessage', $messageWrapper->getMessages()[0]);
     }
 
+    public function testWebChatButtonMessageWithExternalButtons()
+    {
+        OutgoingIntent::create(['name' => 'Hello']);
+        $intent = OutgoingIntent::where('name', 'Hello')->first();
+
+        $generator = new MessageMarkUpGenerator();
+        $buttons = [
+            [
+                'text' => 'Button 1',
+                'callback' => 'callback1',
+                'value' => 'value'
+            ],
+            [
+                'text' => 'Button 2',
+                'callback' => 'callback2',
+                'value' => 'value'
+            ],
+            [
+                'text' => 'Button 3',
+                'callback' => 'callback3',
+                'value' => 'value'
+            ]
+        ];
+        $generator->addButtonMessage('test button', $buttons, true);
+
+        $attributes = ['username' => 'user.name'];
+        $parameters = ['value' => 'dummy'];
+
+        $conditions = new ConditionsYamlGenerator();
+        $conditions->addCondition($attributes, $parameters, 'eq');
+
+        MessageTemplate::create([
+            'name' => 'Friendly Hello',
+            'outgoing_intent_id' => $intent->id,
+            'conditions' => $conditions->getYaml(),
+            'message_markup' => $generator->getMarkUp(),
+        ]);
+
+        // Setup a context to have something to compare against
+        /* @var ContextService $contextService */
+        $userContext = $this->createUserContext();
+        $userContext->addAttribute(new StringAttribute('name', 'dummy'));
+
+        $responseEngineService = $this->app->make(ResponseEngineServiceInterface::class);
+        $messageWrapper = $responseEngineService->getMessageForIntent('webchat', 'Hello');
+
+        $message = $messageWrapper->getMessages()[0];
+
+        $this->assertInstanceOf('OpenDialogAi\ResponseEngine\Message\Webchat\WebchatButtonMessage', $message);
```

Comment:
Please use the static `class` properties here, rather than a hardcoded string.
```suggestion
        $this->assertInstanceOf(WebchatButtonMessage::class, $message);
```